dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -0,0 +1,718 @@
+
+    @property
+    def _boottime(self) -> int:
+        """Returns the boot time of the system.
+
+        Used internally to determine process start- and runtimes.
+        """
+        for line in self.target.fs.get("/proc/stat").open().readlines():
+            if not line.startswith(b"btime"):
+                continue
+
+            return int(line.split()[1])
+
+    @internal
+    def get(self, path) -> FilesystemEntry:
+        """Returns a FilesystemEntry relative to this process."""
+        return self.entry.get(path)
+
+    @property
+    def owner(self) -> str:
+        """Return the username or the uid (if owner is not found) of the owner of this process."""
+        if self.uid:
+            owner = self.target.user_details.find(uid=self.uid)
+            return owner.user.name
+
+        return str(self.uid)
+
+    @property
+    def uid(self) -> int:
+        """Return the uid of the owner of this process."""
+
+        uid = int((self.get("loginuid").open().read()))
+        # loginuid can hold the value "4294967295". Which is defined as "not set" and -1 should be returned.
+        return -1 if c_int32(uid) == -1 else uid
+
+    @property
+    def pid(self) -> int:
+        """Returns the process id (pid) associated to this process."""
+        return self._pid
+
+    @property
+    def parent(self) -> Optional[ProcProcess]:
+        """Returns the parent `ProcProcess of this process."""
+        if self.pid == 0:
+            return None
+        return ProcProcess(self.target, self._stat_file.get("ppid"))
+
+    @property
+    def ppid(self) -> int:
+        """Returns the parent process id (ppid) assiciated to this process."""
+        if self.pid == 0:
+            return None
+        return self.parent.pid
+
+    @property
+    def ppid_name(self) -> str:
+        """Returns the name accociated to the parent process id (ppid) of this process."""
+        return self.parent.name
+
+    @property
+    def state(self) -> str:
+        """Returns the state of the process (S'leeping, R'unning, I'dle, etc)."""
+        return ProcessStateEnum[self._stat_file.get("state")].value
+
+    @property
+    def starttime(self) -> datetime:
+        """Returns the start time of the process."""
+        # Starttime is saved in clockticks per second from the boot time.
+        # we asume a standard of 100 clockticks per second. the actual value can be obtained from `getconf CLK_TCK`
+        starttime = self._stat_file.get("starttime") / 100
+
+        return from_unix(self._boottime + starttime)
+
+    @property
+    def runtime(self) -> timedelta:
+        """Returns the runtime of a process until the moment of acquisition."""
+        runtime = self.now - self.starttime
+        return runtime
+
+    @property
+    def now(self) -> datetime:
+        """Returns the `now()` timestamp of the system at the moment of acquisition."""
+        now = self.uptime + from_unix(self._boottime)
+        return now
+
+    @internal
```

Comment:
```suggestion
    @internal
```
As this is not a `Plugin` subclass, this annotation is not needed.